students/jesse_miller/session04/trigrams.py

```diff
@@ -16,55 +13,62 @@ def parse_file():
     dictionary.
     Thanks for the help Luis!
     """
-    temp_dict = {}
-    sherlock_dict = {}
-    translator = str.maketrans('', '', string.punctuation)
-    sherlock_input = set(line.strip() for line in open('sherlock_small.txt'))
+    startline = 'Produced by an anonymous Project Gutenberg'\
+                'volunteer and Jose Menendez'
+    endline = 'End of the Project Gutenberg EBook'
 
-    for i in sherlock_input:
-        i = i.translate(translator).split(' ')
-        temp_dict = dict(i[l:l+2] for l in range(0, len(i) - 2))
-        sherlock_dict = dict(list(sherlock_dict.items()) +
-                             list(temp_dict.items()))
-    create_tuples(sherlock_dict)
+    with open('sherlock.txt') as input_file:
+        input_file = [line.rstrip('\n') for line in input_file]
+        full_list = []
+        start = 0
+        for line in input_file:
+            if line[:len(startline)] == startline:
+                start = 1
+                continue
+            if not start:
+                input_file = [line.rstrip('\n') for line in input_file]
+            if line[:len(endline)] == endline:
+                break
+    return full_list
```

Comment:
Note that you're returning _full_list_ (an empty list), you could just read the file, remove carriage (which you're doing) and return _input_file_ for _make_words_ to work on it:

```
def parse_file(filename):
    with open(filename) as input_file:
        input_file = [line.rstrip('\n') for line in input_file]
    return input_file
```